### PR TITLE
README tweaks

### PR DESCRIPTION
**Pull request by Netruk44**
Two small tweaks to the README.

1. Update `cond_drop_prob` to be 0.1 instead of 0.5, which is the current default value in the Imagen class. Anecdotally, the current value of 0.5 causes the model to ignore text prompts when sampling images with a `cond_prob` value under 5.0.

2. For simplicity's sake in the examples, leave generating the text encodings to the Imagen and only supply `texts`. The examples don't describe how to generate text encodings from T5, and the Imagen model is currently capable of doing that by itself.

### CODE REVIEW

**Comment by lucidrains** [comment]
@Netruk44 hello! oh nice, so for point 1, this means you already have trained a conditional image generator that is semi-successful?

for point 2, although you are correct, i think eventually to scale up training we need all the text and text masks precomputed (i'll take care of that soon too). i would welcome another section in the readme for the simplified version though (passing in the raw text as a list of strings)

**Comment by Netruk44** [comment]
@lucidrains Yes, I've trained a model that I would say is pretty successful at generating images from a text prompt, except for two issues.

First is that the model ignores the prompt unless I crank the `cond_prob` up to at least 4 or 5, as I've mentioned (but once cond_prob is set high enough, the images conform to the prompt). I've confirmed that this issue isn't unique to me in the LAION discord. Some other people chimed in and suggested that `cond_drop_prob` should be limited to 8-10%, and that 50% is probably way too high. I noticed that you are already using the suggested 10% in the code (just not in the readme), and I'm trying to continue training my model at 10% to see if that resolves the 'ignoring text prompt' issue. So while I don't exactly know if this will fix the issue we're seeing, it seems like a likely candidate glancing over the code.

Second, the final images all have 'swirls' in them \([example 'blank' image](https://cdn.discordapp.com/attachments/979781255270920233/984132834778284092/unknown.png)\). I think this is just a symptom of an untrained upsampler Unet though, as when I generate samples with `stop_at_unet_number = 1` the swirls aren't there at all. I've trained one diffusion model before, so I know that the upsampler takes a really long time to train, so I'm willing to wait a bit more to see if this resolves itself.

I'll add another section for the simplified version shortly.

**Comment by lucidrains** [comment]
@Netruk44 thanks for relaying your results! i'll probably remove the `wip` flag today after i add the linear attention option then

**Comment by lucidrains** [comment]
and yes, you diagnosed the root cause of the observed swirls correctly :)

**Comment by lucidrains** [comment]
@Netruk44 thank!

**Comment by zhangfanTJU** [comment]
@Netruk44 Could you provide your training pipline? Thanks a lot!

**Comment by Netruk44** [comment]
@zhangfanTJU I've been busy with personal stuff for the past month, so I haven't been paying much attention to the developments being made here, it's probable there's better training scripts out there than mine at this point.

If you want my original hacked together training pipeline you can find it here: https://gist.github.com/Netruk44/38d793e6d04a53cc4d9acbfadbb04a5c . As a warning, it's a month old at this point. It's probable you'll need to make some tweaks to get it working with the current state of the repository.

Alternatively, lupinetine took that script and integrated it into this imagen repo, and also made some improvements. You can find that here: https://github.com/lupinetine/imagen-pytorch-lucid

**Comment by zhangfanTJU** [comment]
Thank you very much for your generous help!
